plenario/api/validator.py

```diff
@@ -131,36 +158,178 @@ def validate(validator, request_args):
 
     args = request_args.copy()
 
-    # convert string values to corresponding types so the validator can work
-    convert(args)
-    # returns validated parameters as strings
-    result = validator.dump(args)
-    # convert any remaining strings in args to the right type
-    convert(result.data)
+    # For validator dataset_name__in... need to find a better way to
+    # make it play nice with the validator.
+    if args.get('dataset_name__in'):
+        args['dataset_name__in'] = args['dataset_name__in'].split(',')
 
-    # determine unchecked parameters provided in the request
-    unchecked = set(args.keys()) - set(validator.fields.keys())
+    # This first validation step covers conditions that are dataset
+    # agnostic. These are values can be used to apply to all datasets
+    # (ex. obs_date), or concern the format of the response (ex. limit,
+    # datatype, offset).
+
+    # If there are errors, fail quickly and return.
+    result = validator.load(args)
+    if result.errors:
+        return result
```

Comment:
the first `.convert` I used was redundant. The `.load` method converts the arguments, and has the added benefit of reporting errors early, that the we don't need to continue in the validate function if we don't have to